### ISSUE TITLE
feat: ai vision service — wheat disease detection

### ISSUE THREAD

**Issue by Ahmed-fall** [Ahmed-fall/Silo#6]
## Context
This is a completely independent FastAPI microservice.
It runs on port 8001. It never touches the backend/ folder.
The backend will call it via HTTP after an image is uploaded.

## How it fits in the architecture
Backend (port 8000)
  → POST http://ai-vision:8001/analyze (sends image file)
    → YOLOv8 runs inference
      → returns label + confidence
        → backend saves result to images table

## Step 1 — Training (run offline, not in Docker)
Dataset: https://www.kaggle.com/datasets/freedomfighter1290/wheat-disease

1. Download dataset locally
2. Create train.py in ai-vision/ root (NOT inside app/)
3. Fine-tune YOLOv8n:
   from ultralytics import YOLO
   model = YOLO("yolov8n.pt")
   model.train(data="path/to/data.yaml", epochs=30, imgsz=640)
4. Save best weights to: ai-vision/app/model/weights/best.pt
5. Commit only the weights file, not the dataset

### File: ai-vision/app/model/analyze.py
- Loads weights from app/model/weights/best.pt on startup
- run_inference(image_bytes) → returns { label, confidence }
- If weights file not found: return { "status": "unavailable" }

### File: ai-vision/app/main.py
POST /analyze
- Input: image file (multipart/form-data, field name: "file")
- Calls run_inference()
- Output:
  {
    "label": "rust|blast|mildew|healthy",
    "confidence": 0.91,
    "status": "ok"
  }
- If unavailable:
  { "status": "unavailable" }

GET /health
- Returns { "status": "ok", "service": "ai-vision" }

## File: ai-vision/Dockerfile
Already written. Do not modify it.
if you need to modify it, please contact me on whatsapp

## Rules
- Do NOT import anything from backend/
- Do NOT connect to PostgreSQL
- Never crash — always return a response even on error
- Open PR to main when done, tag mefor review
